source/dub/dependency.d

```diff
@@ -460,11 +486,28 @@ unittest {
 	assert(!a.optional);
 	assert(a.valid);
 	assertThrown(a.version_);
+	assert(a.matches(Version.MASTER));
+	assert(a.matches(Version("1.0.0")));
+	assert(a.matches(Version("0.0.1-pre")));
+	b = Dependency(">=1.0.1");
+	assert(b == a.merge(b));
+	assert(b == b.merge(a));
+	b = Dependency(Version.MASTER);
+	assert(a.merge(m) == m);
+	assert(m.merge(a) == m);
```

Comment:
The idea is that `merge` is commutative. And since `a` is the special `ANY` ("*") constant, the result should always equal the other operand. Probably this should be documented somewhere...